src/Features/header/components/Cities/Cities.tsx

```diff
@@ -1,40 +1,66 @@
 import { FiNavigation } from 'react-icons/fi';
-import { ReactElement, useMemo, useCallback } from 'react';
+import { ReactElement, useMemo, useState, useEffect, useCallback } from 'react';
 
-import { useFetchCityListQuery } from '../Sidebar/SidebarSlice';
-import { LoadingSpinner } from '../../../../components/LoadingSpinner/LoadingSpinner';
+import { LoadingSpinner } from '../../../../components/LoadingSpinner';
+import { useLazyFetchCityListQuery } from '../Sidebar/SidebarSlice';
 import { CityList } from './Cities.types';
 
 export const Cities = (): ReactElement => {
-  const { data, isLoading } = useFetchCityListQuery('');
+  const [page, setPage] = useState(0);
+  const [offset] = useState(8);
+  const [trigger, result] = useLazyFetchCityListQuery();
+
+  const fetchMoreData = useCallback(() => {
+    trigger({ page, offset });
+    setPage((prev) => prev + 1);
+  }, [trigger, offset, page]);
 
   const handleCityClick = useCallback(() => {
-    return data;
-  }, [data]);
+    return result.data;
+  }, [result.data]);
+
+  useEffect(() => {
+    trigger({ page, offset });
+    setPage((prev) => prev + 1);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   return useMemo(
     () => (
-      <div>
-        {isLoading ? (
+      <div className="scroll-smooth" id="scrollableDiv">
+        {result.isLoading ? (
           <LoadingSpinner />
         ) : (
-          <ul>
-            {data?.map(({ city, branch }: CityList) => (
-              <li key={city} className="border-xl" onClick={handleCityClick}>
-                <div className="flex p-2 m-2 font-button_font">
-                  <FiNavigation color="#BFA15C" size={20} fill="#BFA15C" className="mt-4" />
-                  <div className="ml-4">
-                    <p className="font-medium text-gold-title">{city}</p>
-                    <p>{branch} филиалов</p>
+          <>
+            <ul>
+              {result.data?.map(({ id, name, branch }: CityList) => (
+                <li key={id} className="border-xl" onClick={handleCityClick}>
+                  <div className="flex p-2 m-1.5 font-button_font">
+                    <FiNavigation color="#BFA15C" size={20} fill="#BFA15C" className="mt-4" />
+                    <div className="ml-4">
+                      <p className="font-medium text-gold-title">{name}</p>
+                      <p>{branch} филиалов</p>
+                    </div>
                   </div>
-                </div>
-                <hr className="bg-black h-0.5 w-11/12 ml-4" />
-              </li>
-            ))}
-          </ul>
+                  <hr className="bg-black h-0.5 w-11/12 ml-4" />
+                </li>
+              ))}
+            </ul>
+            {result.data?.length >= 38 ? (
```

Comment:
38 - in real world such behavior won't occur. Your can understand that this is the end of the list if your `result.data?.length` is a falsy value, then you show this message. If `.length` is truthy, you show the button